Realm/Tests/QueryTests.m

```diff
@@ -565,6 +565,36 @@ - (void)verifySort:(RLMRealm *)realm column:(NSString *)column ascending:(BOOL)a
     XCTAssertEqualObjects(obj[column], val, @"%@", column);
 }
 
+- (void)testEmbeddedObjectQuery
+{
+    RLMRealm *realm = [self realm];
+    [realm beginWriteTransaction];
+    EmbeddedIntParentObject *obj0 = [EmbeddedIntParentObject createInRealm:realm withValue:@[@1, @[@2], @[@[@3]]]];
+    EmbeddedIntParentObject *obj1 = [EmbeddedIntParentObject createInRealm:realm withValue:@[@4, @[@5], @[@[@6]]]];
+    EmbeddedIntParentObject *obj2 = [EmbeddedIntParentObject createInRealm:realm withValue:@[@7, @[@8], @[@[@9]]]];
+    [realm commitWriteTransaction];
+    
+    NSArray *(^asArray)(RLMResults *) = ^(RLMResults *results) {
+        return [[self evaluate:results] valueForKeyPath:@"self"];
+    };
+    
+    // Query parent objects based on property of embedded object
+    RLMResults *r0 = [EmbeddedIntParentObject objectsWhere:@"object.intCol = 2"];
+    XCTAssertEqualObjects(asArray(r0), (@[ obj0 ]));
```

Comment:
The as array lambda you have there is a bit awkward and there is simpler way to achieve this:
```suggestion
    XCTAssertEqualObjects(r0[0], obj0);
```
You can do this with all of the proceeding equality checks as well.